### PR TITLE
Transaction and checkpoints

### PR DESCRIPTION
**Pull request by delamonpansie**

### CODE REVIEW

File: rocks.ml [hunk 1]
```diff
@@ -215,13 +224,235 @@ module rec Iterator : Rocks_intf.ITERATOR with type db := RocksDb.t = struct
...
+
+  let delete ?(pos=0) ?len ?opts t key =
+    let open Bigarray.Array1 in
+    let len = match len with None -> dim key - pos | Some len -> len in
+    with_err_pointer (delete_raw t (bigarray_start array1 key +@ pos) len)
+
+  let delete_string ?(pos=0) ?len ?opts t key =
+    let len = match len with None -> String.length key - pos | Some len -> len in
+    with_err_pointer (delete_raw_string t (ocaml_string_start key +@ pos) len)
+  let write_raw =
```
Comment (domsj) [inline rocks.ml]: unused?

File: rocks_intf.ml [hunk 2]
```diff
@@ -66,4 +104,10 @@ module type ROCKS = sig
   val write : ?opts:WriteOptions.t -> t -> batch -> unit
 
   val flush : ?opts:FlushOptions.t -> t -> unit
+
+  val create_snapshot : t -> Snapshot.t
+  val relese_snapshot : t -> Snapshot.t -> unit
```
Comment (domsj) [inline rocks_intf.ml]: typo

**Review by domsj — COMMENTED** [review]
Thansk Yuriy, looks good. 2 minor comments, see below

File: rocks.ml (hunk 1 above)
Comment (delamonpansie) [reply rocks.ml]: Yes, sorry.

File: rocks_intf.ml (hunk 2 above)
Comment (delamonpansie) [reply rocks_intf.ml]: thank you.

**Pull request merged**